File: src/TaskService/Entities/Task.cs
namespace TaskService.Entities
{
   public enum TaskStatus
   {
      Unknown,
      Complete,
      Incomplete
   }

   public class Task
   {
      public Guid Id { get; set; }
      public string Name { get; set; }
      public string Description { get; set; }
      public TaskStatus Status { get; set; }
      public bool Disabled { get; set; }
      public DateTime CreatedAt { get; set; }
      public DateTime ModifiedAt { get; set; }
      public bool Deleted { get; set; }
   }
}

File: frontend/client-app/src/models/task-svc/task.ts
import type { TaskStatusCode } from "../../constants/constants";
import { Guid } from "../../types/types";

export interface ITask {
   id: Guid;
   name: string;
   code: string;
   description: string;
   orderIndex: number;
   statusId: Guid;
   statusName: string;
   statusCode: TaskStatusCode;
   disabled: boolean;
   createdAt?: Date;
   modifiedAt?: Date;
   deleted: boolean;
}

export class Task implements ITask {
   id: Guid = Guid.Empty;
   name = "";
   code = "";
   description = "";
   orderIndex = 0;
   statusId: Guid = Guid.Empty;
   statusName = "";
   statusCode: TaskStatusCode = "INCOMPLETE";
   disabled = false;
   createdAt?: Date | undefined;
   modifiedAt?: Date | undefined;
   deleted = false;
}

export const InitialTask: ITask = {
   id: Guid.Empty,
   name: "",
   code: "",
   description: "",
   orderIndex: 0,
   statusId: Guid.Empty,
   statusName: "",
   statusCode: "INCOMPLETE",
   disabled: false,
   createdAt: new Date(),
   modifiedAt: new Date(),
   deleted: false,
};
